src/fill-in-the-blank/components/runtime.test.tsx

```diff
@@ -1,110 +1,168 @@
+ * Enzyme doesn't work with components that use dangerouslySetInnerHtml().
+ */
+
 describe("Runtime", () => {
+
   it("renders prompt and inputs", () => {
-    const wrapper = shallow(<Runtime authoredState={authoredState} />);
-    expect(wrapper.text()).toEqual(expect.stringContaining("Test prompt with "));
-    expect(wrapper.find("input").length).toEqual(2);
-    expect(wrapper.find("input").at(0).props().size).toEqual(10);
-    expect(wrapper.find("input").at(1).props().size).toEqual(20);
+    const { container, getByText } = render(<Runtime authoredState={authoredState} />);
+    expect(getByText("Test prompt with", { exact: false })).toBeDefined();
+    const inputs = container.querySelectorAll("input");
+    expect(inputs.length).toBe(2);
+    expect(inputs[0].id).toBe("[blank-1]");
+    expect(inputs[0].size).toBe(10);
+    expect(inputs[1].id).toBe("[blank-2]");
+    expect(inputs[1].size).toBe(20);
   });
 
   it("handles passed interactiveState", () => {
-    const wrapper = shallow(<Runtime authoredState={authoredState} interactiveState={interactiveState} />);
-    expect(wrapper.find("input").at(0).props().value).toEqual(interactiveState.blanks[0].response);
-    expect(wrapper.find("input").at(1).props().value).toEqual("");
+    const { container } = render(<Runtime authoredState={authoredState} interactiveState={interactiveState} />);
+    const inputs = container.querySelectorAll("input");
+    expect(inputs[0].value).toEqual(interactiveState.blanks[0].response);
+    expect(inputs[1].value).toEqual("");
+  });
+
+  it("renders minimal states", () => {
+    const minimalAuthoredState: IAuthoredState = { version: 1, questionType: "iframe_interactive" };
+    const minimalInteractiveState: IInteractiveState = { answerType: "interactive_state" } as any;
+    const { container } = render(<Runtime authoredState={minimalAuthoredState} interactiveState={minimalInteractiveState} />);
+    const inputs = container.querySelectorAll("input");
+    expect(inputs.length).toBe(0);
   });
 
   it("calls setInteractiveState when user provides an answer", () => {
     const setState = jest.fn();
-    const wrapper = shallow(<Runtime authoredState={authoredState} interactiveState={interactiveState} setInteractiveState={setState} />);
-    wrapper.find("input").at(1).simulate("change", { target: { value: "New response" } });
+    const { container } = render(<Runtime authoredState={authoredState} interactiveState={interactiveState} setInteractiveState={setState} />);
+    const inputs = container.querySelectorAll("input");
+    fireEvent.change(inputs[1], { target: { value: "New response" } });
     const newState = setState.mock.calls[0][0](interactiveState);
     expect(newState).toEqual({
       answerType: "interactive_state",
-      answerText: "Test prompt with [ Test response ] and [ New response ].",
+      answerText: "<p>Test prompt with [ Test response ] and [ New response ].</p>",
```

Comment:
>It'd be nice if Cypress tests don't stay like that forever. ;) But I guess they can be fixed in other PR, depends on timing and priorities. Probably some PT story would be nice so it doesn't get lost.

Agreed, but I don't know how much work is involved in getting cypress tests to work with the RTE so it seems reasonable to put it off to another PR given this one has already gotten quite large.

>The second thing would be HTML in `answerText` - I feel like we should avoid it at this point, but maybe @scytacki has some opinion on that too (Teacher Report could render HTML if we want, and it could have some benefits too).

Good point -- I suspect you're right. We'd have to strip the tags from the prompt in that case.

>I'm wondering about using `dangerouslySetInnerHTML()` everywhere. Aren't there available some libraries that parse HTML and generate React components? I could imagine that they could also deal with incorrect HTML and escape dangerous parts (since Runtime doesn't know where the HTML is coming from in fact). I googled quickly and saw: https://www.npmjs.com/package/html-react-parser . This could also fix problems with Enzyme.

I think that's worth an experiment. As you say it would be nice not to have to use `dangerouslySetInnerHTML()`. It looks like html-react-parser doesn't specifically deal with security issues. I wondered about using `DOMPurify` on the interactive side as well but the risks seemed much less in an interactive iframe than on the LARA side, for instance, so I wasn't sure if it was worth it. I think I prefer @testing-library/react at this point, however, so I'm not switching the tests back to Enzyme even if it works. 😉

Reply:
I've added some additional commits which:
- convert the `answerText` to plain text
- replace use of `dangerouslySetInnerHTML()` with `html-react-parser` and `DOMPurify`